### PR TITLE
Story/select fire zone/2545

### PR DESCRIPTION
**Pull request by andrea-williams**
- On initial page load, no fire zone is selected. Map is full-width of browser window
- Once fire zones have loaded on map (zone labels won't render until zoomed in), user can click on a fire zone on the map to select it
- When a fire zone has been selected, the `ZoneSummaryPanel` will render on the left side of the page. For now the only thing in the panel is the ID number of the selected fire zone
- When a fire zone has been selected, the map will automatically resize/shift so that the extent of the fire zone fills the map view. Right now this isn't perfect because our map has horizontal & vertical scroll bars, but we have an upcoming bug ticket to fix that so that the entire fire zone should be visible on the same page without any additional adjustment required by the user.
- When a fire zone has been selected, the border of the fire zone will be rendered in green and the fire zone label will be rendered in green, larger font than the rest of the fire zones. (This re-rendering doesn't happen instantaneously, you gotta wait a little bit).
- Note: current behaviour is that once a fire zone has been selected, you can't de-select a fire zone (but you can select a different one).

example:
Screenshot 2023-01-30 at 4 43 49 PM

# Test Links:
[Percentile Calculator](https://wps-pr-2572.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-2572.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-2572.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-2572.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-2572.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-2572.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-2572.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: web/src/features/fba/components/ZoneSummaryPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+const useStyles = makeStyles({
+  autocomplete: {
+    width: '100%',
+    color: 'white'
+  },
+  wrapper: {
+    minWidth: 300
+  },
+  fireCentreTextField: {
```
Comment (conbrad) [inline web/src/features/fba/components/ZoneSummaryPanel.tsx:14]: Unused?

File: web/src/features/fba/components/ZoneSummaryPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        borderColor: 'white'
+      },
+      '&.Mui-focused fieldset': {
+        borderColor: 'white'
+      }
+    }
+  }
+})
+
+export interface Option {
```
Comment (conbrad) [inline web/src/features/fba/components/ZoneSummaryPanel.tsx:43]: Unused?

File: web/src/features/fba/components/map/FBAMap.tsx [hunk 3]
```diff
@@ -120,38 +141,55 @@ const FBAMap = (props: FBAMapProps) => {
...
+          }
+          props.setSelectedFireZoneID(feature.get('mof_fire_zone_id'))
+        })
+      })
     }
-  }, [map, fireZoneVector])
+  }, [map]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    fetchFireZoneAreas(props.runType, props.runDate.toISO(), props.forDate.toISODate())
```
Comment (conbrad) [inline web/src/features/fba/components/map/FBAMap.tsx:167]: The current pattern is the page will do all the fetching calls to keep FBAMap isolated with its own local state but I can adjust this in the advisory color PR

**Review by conbrad — APPROVED** [review]
🚢

File: web/src/features/fba/components/ZoneSummaryPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { TextField } from '@mui/material'
+import makeStyles from '@mui/styles/makeStyles'
+import { isNull } from 'lodash'
+
+const useStyles = makeStyles({
+  autocomplete: {
```
Comment (dgboss) [inline web/src/features/fba/components/ZoneSummaryPanel.tsx:7]: Looks like autocomplete is unused as well.

**Review by dgboss — APPROVED** [review]

**Pull request merged**